development/tools/pkg/prtagbuilder/prtagbuilder.go

```diff
@@ -12,6 +13,37 @@ import (
 	"k8s.io/test-infra/prow/pod-utils/downwardapi"
 )
 
+type githubRepoService interface {
+	GetBranch(ctx context.Context, owner, repo, branch string) (*github.Branch, *github.Response, error)
+	GetCommit(ctx context.Context, owner, repo, sha string) (*github.RepositoryCommit, *github.Response, error)
+	// other repository methods used in your app
+}
+
+type githubPullRequestsService interface {
+	Get(ctx context.Context, owner string, repo string, number int) (*github.PullRequest, *github.Response, error)
+	// other pull requests methods used in your app
+}
+
+// GitHubClient is an implementation of go-github client with needed services only.
+type GitHubClient struct {
+	Repositories githubRepoService
+	PullRequests githubPullRequestsService
+	// optionally store and export the underlying *github.Client
+	// if you want easy access to client.Rate or other fields
+}
+
+// NewGhClient returns new instance of go-github GitHubClient implementation.
+func NewGhClient(httpClient *http.Client) *GitHubClient {
```

Comment:
It might be a little bit picky but if the struct is called the `GitHubClient` the New function should have the full name of a struct - `NewGitHubClient()`. This way it should be easy to notice what this function does and returns. Otherwise this suggestion fully optional and can be omitted.